DQMOffline/Trigger/plugins/ZGammaplusJetsMonitor.cc

```diff
@@ -0,0 +1,814 @@
+                             std::abs(muon_2.Eta()) < muon_eta;
+            if (!muon_pass) {
+              return;
+            }
+
+            Zhltreco = muon_1 + muon_2;
+            bool Z_pass = std::abs(Zhltreco.M() - 91.2) < Z_DM && Zhltreco.Pt() > Z_Pt;
+            if (!Z_pass) {
+              return;
+            }
+            trigobj_pt.push_back(Zhltreco.Pt());
+            trigobj_eta.push_back(Zhltreco.Eta());
+            trigobj_phi.push_back(Zhltreco.Phi());
+          }       //end check di-object
+          else {  //if not di-object
+            return;
+          }
+        }  // end avoid duplicate objects
+        kCnt1++;
+      }  // key1
+      kCnt0++;
+    }  // key
+  }    // end else
+
+  // ---------------- module for Jet leg Jets --------------------------
+  // index of last module executed in this Path
+  const unsigned int moduleIndex = triggerResults->index(index);  // here would be HLTBool at the end
+  edm::LogInfo("ZGammaplusJetsMonitor")
+      << " Module Index " << moduleIndex - 1 << " Module Name  "
+      << module_names[moduleIndex - 1];  // the second to last would be the last module that is saved
+  assert(moduleIndex < module_size);
+
+  // results from TriggerEvent product
+  const std::string& ImoduleLabel = module_names[moduleIndex - 1];
+  const std::string ImoduleType = hltConfig_.moduleType(ImoduleLabel);
+  edm::LogInfo("ZGammaplusJetsMonitor") << ImoduleLabel << " |  " << ImoduleType;
+  // check whether the module is packed up in TriggerEvent product
+  const unsigned int filterIndex = aodTriggerEvent->filterIndex(edm::InputTag(ImoduleLabel, "", processName_));
+  if (filterIndex >= aodTriggerEvent->sizeFilters()) {
+    return;
+  }
+  const trigger::Vids& VIDS = aodTriggerEvent->filterIds(filterIndex);
+  const trigger::Keys& KEYS = aodTriggerEvent->filterKeys(filterIndex);
+  const trigger::size_type nI = VIDS.size();
+  const trigger::size_type nK = KEYS.size();
+  assert(nI == nK);
+  const trigger::TriggerObjectCollection& objects(aodTriggerEvent->getObjects());
+  for (trigger::size_type idx = 0; idx < nI; ++idx) {
+    const trigger::TriggerObject& TO_(objects[KEYS[idx]]);
+    //VIDS :  muons-->83 / jets-->85  / met-->87
+    edm::LogInfo("ZGammaplusJetsMonitor")
+        << " idx  " << idx << " vid  " << VIDS[idx] << "/"
+        << " keys  " << KEYS[idx] << ": "
+        << " obj_id " << TO_.id() << " " << TO_.pt() << " " << TO_.eta() << " " << TO_.phi() << " " << TO_.mass();
+  }
+  for (const auto& key : KEYS) {
+    v_jetpt.push_back(objects[key].pt());
+    v_jeteta.push_back(objects[key].eta());
+    v_jetphi.push_back(objects[key].phi());
+  }
+  bool Jet_pass = (!v_jetpt.empty() && v_jetpt[0] >= pt_cut);
+  if (!Jet_pass) {
+    return;
+  }
+  double dphi = std::abs(v_jetphi[0] - trigobj_phi[0]);
+  if (dphi > acos(-1.)) {
+    dphi = 2 * acos(-1.) - dphi;
+  }
```

Comment:
```suggestion
  if (dphi > M_PI) {
    dphi = 2 * M_PI - dphi;
  }
```

??

Reply:
Sorry I forgot dR2 the first time!! Thanks for the comments!